ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -560,9 +592,9 @@ private void fetchRequests(RequestRegistrationCallback<GetOperation> requestRegi
      * Check if the operation on the chunk is eligible for completion, if so complete it.
      */
     void checkAndMaybeComplete() {
-      if (chunkOperationTracker.isDone()) {
+      if (!chunkCompleted && chunkOperationTracker.isDone()) {
```

Comment:
`chunkCompleted` seems a little overloaded and confusing. Perhaps this can be refactored a little to make it more understandable?